src/printer/tvmscript_unified_printer.cc

```diff
@@ -0,0 +1,738 @@
+    } else if (doc->IsInstance<TypeCallDocNode>()) {
+      PrintDoc(Downcast<TypeCallDoc>(doc));
+    } else if (doc->IsInstance<FunctionDocNode>()) {
+      PrintDoc(Downcast<FunctionDoc>(doc));
+    } else if (doc->IsInstance<FunctionArgDocNode>()) {
+      PrintDoc(Downcast<FunctionArgDoc>(doc));
+    } else {
+      LOG(FATAL) << "Do not know how to print " << doc->GetTypeKey();
+    }
+  };
+
+  virtual void PrintDoc(const LiteralValueDoc& doc) = 0;
+  virtual void PrintDoc(const ConstDoc& doc) = 0;
+  virtual void PrintDoc(const IdentifierDoc& doc) = 0;
+  virtual void PrintDoc(const AttrAccessDoc& doc) = 0;
+  virtual void PrintDoc(const IndexDoc& doc) = 0;
+  virtual void PrintDoc(const OperationDoc& doc) = 0;
+  virtual void PrintDoc(const CallDoc& doc) = 0;
+  virtual void PrintDoc(const TupleDoc& doc) = 0;
+  virtual void PrintDoc(const SeqStmtDoc& doc) = 0;
+  virtual void PrintDoc(const ScopeDoc& doc) = 0;
+  virtual void PrintDoc(const ForDoc& doc) = 0;
+  virtual void PrintDoc(const AssignDoc& doc) = 0;
+  virtual void PrintDoc(const ExprTypeDoc& doc) = 0;
+  virtual void PrintDoc(const TypeCallDoc& doc) = 0;
+  virtual void PrintDoc(const FunctionDoc& doc) = 0;
+  virtual void PrintDoc(const FunctionArgDoc& doc) = 0;
+
+  using OutputStream = std::ostringstream;
+
+  OutputStream& NewLine() {
+    output_ << "\n" << std::string(indent_, ' ');
+    return output_;
+  }
+
+  OutputStream output_;
+  int indent_ = 0;
+};
+
+class PythonDocPrinter : public DocPrinter {
+ public:
+  PythonDocPrinter(String tir_prefix) : tir_prefix_(tir_prefix) {}
+
+ protected:
+  using DocPrinter::PrintDoc;
+
+  virtual void PrintDoc(const LiteralValueDoc& doc) override {
+    auto& value = doc->value;
+    if (value->IsInstance<FloatImmNode>() || value->IsInstance<IntImmNode>()) {
+      PrintNumberNode(Downcast<PrimExpr>(doc->value));
+    } else if (value->IsInstance<StringObj>()) {
+      PrintStringLiteral(Downcast<String>(value));
+    } else if (value->IsInstance<tir::StringImmNode>()) {
+      PrintStringLiteral(Downcast<StringImm>(value)->value);
+    }
+  };
+
+  virtual void PrintDoc(const ConstDoc& doc) override {
+    switch (doc->kind) {
+      case ConstDocNode::ConstKind::TIRBuilder:
+        output_ << tir_prefix_;
+        break;
+      case ConstDocNode::ConstKind::RelaxBuilder:
+        LOG(FATAL) << "RelaxBuilder constant not supported yet";
+        break;
+      case ConstDocNode::ConstKind::None:
+        output_ << "None";
+        break;
+    }
+  };
+
+  virtual void PrintDoc(const IdentifierDoc& doc) override { output_ << doc->name; };
+
+  virtual void PrintDoc(const AttrAccessDoc& doc) override {
+    PrintDoc(doc->value);
+    output_ << ".";
+    PrintDoc(doc->attr);
+  };
+
+  virtual void PrintDoc(const IndexDoc& doc) override {
+    PrintDoc(doc->value);
+    PrintJoinedElements("[", doc->indices, ", ", "]", "()");
+  };
+
+  virtual void PrintDoc(const OperationDoc& doc) override {
```

Comment:
enum class OperationKind : int32_t {
  kBeginBinaryBuiltin = 0,
  kAdd = 0,
  kSub = 1,
  ...
  kFloorMod = 9,
  kEndBinaryBuiltin = 9,
  kBeginBinaryCall = 10,
  kMax = 10,  # "max(a, b)" instead of "a max b",
};

static std::string repr[] = {
  "+", "-", ...
};